extensions/amp-list/0.1/amp-list.js

```diff
@@ -387,15 +398,45 @@ export class AmpList extends AMP.BaseElement {
           AmpEvents.DOM_UPDATE, /* detail */ null, {bubbles: true});
       this.container_.dispatchEvent(event);
 
-      // Change height if needed.
-      this.measureElement(() => {
-        const scrollHeight = this.container_./*OK*/scrollHeight;
-        const height = this.element./*OK*/offsetHeight;
-        if (scrollHeight > height) {
-          this.attemptChangeHeight(scrollHeight).catch(() => {});
-        }
-      });
+      if (!resizeExperimentOn) {
+        // Change height if needed.
+        this.measureElement(() => {
+          const scrollHeight = this.container_./*OK*/scrollHeight;
+          const height = this.element./*OK*/offsetHeight;
+          if (scrollHeight > height) {
+            this.attemptChangeHeight(scrollHeight).catch(() => {});
+          }
+        });
+      }
     }, this.container_);
+
+    // This needs to be gated behind the experiment 'amp-list-resize'
+    if (resizeExperimentOn && !this.isSetToLayoutContainer_) {
+      this.mutateElement(() => {
+        this.changeToLayoutContainer_();
+      });
+      this.isSetToLayoutContainer_ = true;
+    }
+  }
+
+  /**
+   * Converts the amp-list to de facto layout container.
+   * @private
+   */
+  changeToLayoutContainer_() {
+    this.element.setAttribute('layout', 'container');
+    const sizer = childElementByTag(this.element, 'i-amphtml-sizer');
+    if (sizer) {
+      this.element.removeChild(sizer);
+    }
+    this.element.classList.remove(
+        'i-amphtml-layout-size-defined', 'i-amphtml-layout-responsive');
+    this.container_.classList.remove(
+        'i-amphtml-fill-content', 'i-amphtml-replaced-content');
+    const overflowElement = childElementByAttr(this.element, 'overflow');
+    if (overflowElement) {
+      toggle(overflowElement, false);
```

Comment:
Unfortunately, this doesn't work for the same reason I mentioned to Ali above. The overflow element is hidden via `visibility: hidden`, but it takes up a blank space, which looks odd since we are now in layout container. Technically after we switch to layout container, we shouldn't need the `overflow` element anymore...

Reply:
Gotcha, thanks. Can you add a comment explaining this? Also use `getOverflowElement()`.

Reply:
Done